static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt

```diff
@@ -143,19 +144,89 @@ class AutoDisposeDetector : Detector(), SourceCodeScanner {
   override fun createUastHandler(context: JavaContext): UElementHandler? {
     return object : UElementHandler() {
       override fun visitCallableReferenceExpression(node: UCallableReferenceExpression) {
-        val method = node.resolve()
-        // Check if the resolved call reference is method and check that it's invocation is a
-        // call expression so that we can get it's return type etc.
-        if (method is PsiMethod && node.uastParent != null && node.uastParent is UCallExpression) {
-          evaluateMethodCall(node.uastParent as UCallExpression, method, context)
+        node.resolve()?.let { method ->
+          if (method is PsiMethod) {
+            callableReferenceChecker(context, node, method, ::containingClassScopeChecker)
+          }
         }
       }
 
       override fun visitCallExpression(node: UCallExpression) {
-        node.resolve()?.let {
-          evaluateMethodCall(node, it, context)
+        node.resolve()?.let { method ->
+          // Check if it's one of our withScope() higher order functions. If so, we handle that
+          // separately and visit the passed in lambda body and run the subscribe method call checks
+          // inside it with the "isInScope" check just hardcoded to true.
+          if (method.name == "withScope" &&
+              method.containingClass?.qualifiedName == KOTLIN_EXTENSIONS) {
+            val args = node.valueArguments
+            if (args.size == 2) {
+              val last = args[1]
```

Comment:
Can also do `args.last()` i believe.

Reply:
That does an extra size check, which we can skip in this case